templates/user/profile.tmpl

```diff
@@ -16,8 +16,17 @@
 							<p>{{ctx.Locale.Tr "user.disabled_public_activity"}}</p>
 						</div>
 					{{end}}
-					{{template "user/heatmap" .}}
-					{{template "user/dashboard/feeds" .}}
+					{{if or .HeatmapData (not .ContextUser.KeepActivityPrivate)}}
```

Comment:
Why this new condition `not .ContextUser.KeepActivityPrivate` here? Seems like a unrelated change.

Reply:
Because even without active content, the heatmap will still be displayed if the user has not hidden activity on their personal page, just like the empty state added to the dashboard

Reply:
I think it would be better to move this logic out of the template and into the backend place where `.HeatmapData` is being set.

The less logic we have in template, the better because this code would crash the renderer if `.ContextUser` is not there and by moving into backend we have type safety that the templates lack.

Reply:
I agree, but I'm not familiar with the backend, could you help me if possible？

Reply:
Start by looking at these two places:

```
routers/web/user/profile.go:71:		ctx.Data["HeatmapData"] = data
routers/web/user/home.go:111:		ctx.Data["HeatmapData"] = data
```

Reply:
Since this judgment only appears in the `profile.tmpl`, I added some  information in `profile.go`. However, I'm not good at the backend, so I'm not sure if this way of writing is acceptable. As far as function is concerned, there should be no problem as I have tested it.